### ISSUE TITLE
Indentation of annotations and initial equation

### ISSUE THREAD

**Issue by raulrpearson** [modelica-tools/modelica-mode#2]
I'd really like to use Emacs to edit and clean up Modelica files. I'm only using [modelica-mode.el](https://github.com/modelica-tools/modelica-mode/blob/master/modelica-mode.el), but I have some issues with indentation.

Using [an example](https://github.com/xogeny/ModelicaBook/blob/master/ModelicaByExample/BasicEquations/SimpleExample/FirstOrderExperiment.mo) from Modelica by Example, this is what I'd expect to get:

```Modelica
within ModelicaByExample.BasicEquations.SimpleExample;
model FirstOrderExperiment "Defining experimental conditions"
  Real x "State variable";
initial equation
  x = 2 "Used before simulation to compute initial values";
equation
  der(x) = 1-x "Drives value of x toward 1.0";
  annotation(
    experiment(
      StartTime=0,
      StopTime=8));
end FirstOrderExperiment;
```

This is what I get instead:

```Modelica
within ModelicaByExample.BasicEquations.SimpleExample;
model FirstOrderExperiment "Defining experimental conditions"
  Real x "State variable";
  initial equation
  x = 2 "Used before simulation to compute initial values";
equation
  der(x) = 1-x "Drives value of x toward 1.0";
  annotation(
	     experiment(
			StartTime=0,
			StopTime=8));
end FirstOrderExperiment;
```

Notice:
- `initial equation` gets pushed 2 spaces forward when presumably it shouldn't.
- Annotations (or things that use brackets, I guess) get indented to align with enclosing brackets. This isn't incorrect per se, but it quickly makes for very long lines in annotations, which are difficult to read since they end up truncated or wrapped, creating a very unsightly source code.
- We end up with a mixture of tabs and spaces for indentation.

Is this a problem with my setup or is this expected behaviour? I don't mind putting in some time to provide fixes or enhancements if someone provides some pointers, since I'm not familiar with Elisp. Thanks!

**Comment by raulrpearson** [comment]
I think I fixed the the indentation problem with `initial equation` by adding to the regular expression in [line 496](https://github.com/modelica-tools/modelica-mode/blob/master/modelica-mode.el#L496). I haven't pushed to [my fork](https://github.com/raulrpearson/modelica-mode) yet, but will do soon.

**Comment by dietmarw** [comment]
This sounds like an interesting fix. I can not think the current behaviour is intended. So looking forward to your PR :smile:

**Comment by raulrpearson** [comment]
Fair enough, I'll post it as soon as I get it. It's on hold right now because I was spending too much time customizing Emacs and too little time on my Modelica code 😄. I managed to improve it somewhat by changing some of the logic in [mdc-calculate-indent](https://github.com/modelica-tools/modelica-mode/blob/master/modelica-mode.el#L459), but it's not quite there yet.

**Comment by JKRT** [comment]
@raulrpearson 

I maintain a heavily modified fork of modelica-mode (You can really call it XModelica-mode I guess 🖌). Which I finally decided to update yesterday. It differs quite a bit from the original from where it is derived this :D Awesome work btw @maintainers! From insights I know I have a few users and subforks of my own. 

I do actually support the functionality you desire and some more things. 

```
`within ModelicaByExample.BasicEquations.SimpleExample;
model FirstOrderExperiment "Defining experimental conditions"
  Real x "State variable";
initial equation
  x = 2 "Used before simulation to compute initial values";
equation
  der(x) = 1-x "Drives value of x toward 1.0";
  annotation(
             experiment(
             StartTime=0,
             StopTime=8));
end FirstOrderExperiment;`
```

Sadly I indent annotations in much the same way, but initial equation e.t.c s are working as they should : ). If you are interested you can fork mine, put up any issues you have and so on : )

**Comment by raulrpearson** [comment]
Good to hear, @JKRT. If you're running a fork, have you considered submitting a PR to merge your contributions?

I'm not doing much Modelica these days, otherwise I'd give your fork a try. Thanks for letting me know.

**Comment by JKRT** [comment]
> 
> 
> Good to hear, @JKRT. If you're running a fork, have you considered submitting a PR to merge your contributions?
> 
> I'm not doing much Modelica these days, otherwise I'd give your fork a try. Thanks for letting me know.

It is pretty much its own thing by know, and I usually make quite steep changes and break it from time to time. It differs too much to really be mergeable with the original.
